src/main/java/beam/agentsim/events/handling/BeamEventsWriterXML.java

```diff
@@ -48,22 +52,44 @@ public void init(final String outfilename) {
     public void reset(final int iter) {
     }
 
+    //Modify the method to control logger and based on logging level writing specific attribute
     @Override
     protected void writeEvent(final Event event) {
-        try {
-            this.out.append("\t<event ");
-            Map<String, String> attr = event.getAttributes();
-            for (Map.Entry<String, String> entry : attr.entrySet()) {
-                this.out.append(entry.getKey());
-                this.out.append("=\"");
-                this.out.append(encodeAttributeValue(entry.getValue()));
-                this.out.append("\" ");
-            }
-            this.out.append(" />\n");
+        String LoggerLevel=this.beamServices.beamConfig().beam().outputs().defaultLoggingLevel();
+
+        if(!LoggerLevel.equals(LoggerLevels.OFF.toString())&& !LoggerLevel.equals("")){
+
+            List loggerinfotype= null;
+            try {
+                if(LoggerLevel.equals(LoggerLevels.VERBOSE.toString())) {
+                    loggerinfotype = this.beamServices.beamConfig().beam().outputs().VERBOSELoggingLevel();
+                }
+
+                else if(LoggerLevel.equals(LoggerLevels.REGULAR.toString())) {
+                    loggerinfotype = this.beamServices.beamConfig().beam().outputs().REGULARLoggingLevel();
+                }
+
+                else if(LoggerLevel.equals(LoggerLevels.SHORT.toString())) {
+                    loggerinfotype = this.beamServices.beamConfig().beam().outputs().SHORTLoggingLevel();
+                }
+
+                this.out.append("\t<event ");
+                Map<String, String> attr = event.getAttributes();
+                for (Map.Entry<String, String> entry : attr.entrySet()) {
+
+                    if(loggerinfotype.contains(entry.getKey())) {
```

Comment:
this doesn't cover the case when we wrap a MATSim event in our own, but it's a good start.